great_expectations/render/renderer_configuration.py

```diff
@@ -95,49 +97,81 @@ class Config:
         validate_assignment = True
         arbitrary_types_allowed = True
 
-    def __init__(self, **kwargs) -> None:
-        kwargs = RendererConfiguration._set_expectation_type_and_kwargs(kwargs=kwargs)
-        kwargs = RendererConfiguration._set_include_column_name_and_styling(
-            kwargs=kwargs
-        )
-        kwargs["params"] = _RendererParamsBase()
-        super().__init__(**kwargs)
-
-    @staticmethod
-    def _set_expectation_type_and_kwargs(kwargs: dict) -> dict:
-        if "configuration" in kwargs and kwargs["configuration"]:
-            kwargs["expectation_type"] = kwargs["configuration"].expectation_type
-            kwargs["kwargs"] = kwargs["configuration"].kwargs
-        elif (
-            "result" in kwargs
-            and kwargs["result"]
-            and kwargs["result"].expectation_config
+    @root_validator(pre=True)
+    def validate_configuration_or_result(cls, values: dict) -> dict:
```

Comment:
1) I'm only able to get validator to occur if I add `@classmethod` AFTER `@root_validator()`, just FYI
2) Once I do that PyCharm complains: about `@rootvalidator`: `This decorator will not receive a callable it may expect; the built-in decorator returns a special object`

Why doesn't the Pydantic documentation use the `@classmethod` decorator?

Reply:
1. Yes, order often matters with decorators.
2. I've never seen that. But I'm not using PyCharm.

>Why doesn't the Pydantic documentation use the @classmethod decorator?

I don't know, I hadn't noticed that before. They must be making these methods `classmethods` as part of the pydantic metaclass hook. Otherwise, it wouldn't work.